boot/zephyr/serial_adapter.c

```diff
@@ -157,17 +157,23 @@ boot_uart_fifo_getline(char **line)
 {
 	static struct line_input *cmd;
 	sys_snode_t *node;
+	int key;
 
+	key = irq_lock();
```

Comment:
looks like this lock doesn't lock the uart irq at all. code works without irq_loc as well.